recipe/0017-Check-for-changes-then-forcibly-mv-in-javareconf.in.patch

```diff
@@ -1,28 +1,27 @@
-From b15c503e71e0f2edfc1c3ffeb6813bf38ab92ed8 Mon Sep 17 00:00:00 2001
-From: Ray Donnelly <mingw.android@gmail.com>
-Date: Tue, 2 Oct 2018 10:48:53 +0100
-Subject: [PATCH 17/17] Foribly remove then forcibly mv in javareconf.in
+commit 3520c9d3a4fd1f2363c8bbcd268ec42b0be7071a
+Author: Marcel Bargull <marcel.bargull@udo.edu>
+Date:   Sun Jun 16 23:10:00 2019 +0200
 
----
- src/scripts/javareconf.in | 5 ++++-
- 1 file changed, 4 insertions(+), 1 deletion(-)
+    [PATCH 17/17] Check for changes then forcibly mv in javareconf.in
 
 diff --git a/src/scripts/javareconf.in b/src/scripts/javareconf.in
-index d23fcfb47d..33b92c4b8c 100644
+index 5dbe3a3e9a..14cdbc153d 100644
 --- a/src/scripts/javareconf.in
 +++ b/src/scripts/javareconf.in
-@@ -421,7 +421,10 @@ fi
+@@ -402,13 +402,15 @@ else
+   files="${R_HOME}/etc/Makeconf ${R_HOME}/etc/ldpaths"
+ fi
  for file in $files; do
++    ${SED-sed} -e "s|JAVA =.\{0,\}|JAVA = $JAVA|" -e "s|JAVA_HOME =.\{0,\}|JAVA_HOME = ${JAVA_HOME}|" -e "s|: \${JAVA_HOME=.\{1,\}|: \${JAVA_HOME=${JAVA_HOME}}|" -e "s|: \${R_JAVA_LD_LIBRARY_PATH=.\{1,\}|: \${R_JAVA_LD_LIBRARY_PATH=${JAVA_LD_LIBRARY_PATH_SH}}|" -e "s|JAVA_LIBS =.\{0,\}|JAVA_LIBS = ${JAVA_LIBS}|g" -e "s|JAVA_LD_LIBRARY_PATH =.\{0,\}|JAVA_LD_LIBRARY_PATH = ${JAVA_LD_LIBRARY_PATH}|" -e "s|JAVAC =.\{0,\}|JAVAC = $JAVAC|" -e "s|JAVAH =.\{0,\}|JAVAH = $JAVAH|" -e "s|JAR =.\{0,\}|JAR = $JAR|" -e "s|JAVA_CPPFLAGS =.\{0,\}|JAVA_CPPFLAGS = ${JAVA_CPPFLAGS}|g"  | cmp -s - "${file}" || {
      ${SED-sed} -e "s|JAVA =.\{0,\}|JAVA = $JAVA|" -e "s|JAVA_HOME =.\{0,\}|JAVA_HOME = ${JAVA_HOME}|" -e "s|: \${JAVA_HOME=.\{1,\}|: \${JAVA_HOME=${JAVA_HOME}}|" -e "s|: \${R_JAVA_LD_LIBRARY_PATH=.\{1,\}|: \${R_JAVA_LD_LIBRARY_PATH=${JAVA_LD_LIBRARY_PATH_SH}}|" -e "s|JAVA_LIBS =.\{0,\}|JAVA_LIBS = ${JAVA_LIBS}|g" -e "s|JAVA_LD_LIBRARY_PATH =.\{0,\}|JAVA_LD_LIBRARY_PATH = ${JAVA_LD_LIBRARY_PATH}|" -e "s|JAVAC =.\{0,\}|JAVAC = $JAVAC|" -e "s|JAVAH =.\{0,\}|JAVAH = $JAVAH|" -e "s|JAR =.\{0,\}|JAR = $JAR|" -e "s|JAVA_CPPFLAGS =.\{0,\}|JAVA_CPPFLAGS = ${JAVA_CPPFLAGS}|g"  "${file}" > "${file}.new"
      if test -f "${file}.new"; then
 -	mv "${file}.new" "${file}"
-+	if test -f "${file}"; then
-+	    rm -rf "${file}" >/dev/null 2>&1
```

Comment:
@mingwandroid, in which cases do you think the `rm -rf ... ; mv -f` is necessary, rather than just a simple `mv -f`?

Reply:
Should be fine. What do you think to using instead of .new using .$$? doesn't work so well in these docker days sometimes but still worth doing too I think?

Reply:
I'm not sure. I'll have to think about it for a minute. I guess it should be fine.

Reply:
Yeah, I'll update it to use `$$`. No harm in doing so, I guess.